ast/compile.go

```diff
@@ -4798,6 +4798,22 @@ func rewriteWithModifier(c *Compiler, f *equalityFactory, expr *Expr) ([]*Expr,
 
 func validateWith(c *Compiler, expr *Expr, i int) (bool, *Error) {
 	target, value := expr.With[i].Target, expr.With[i].Value
+
+	// Ensure that values that are built-ins are rewritten to Ref (not Var)
+	if v, ok := value.Value.(Var); ok {
+		if _, ok := c.builtins[v.String()]; ok {
+			value.Value = Ref([]*Term{NewTerm(v)})
+		}
+	}
+
+	// For topdown to be able to tell if a replaced data ref is a rule, or
```

Comment:
This is wrong, I think. We can use the rule tree from the e.compiler and that's what we should do. I'm update and clean this up next.